pytest_structlog/__init__.py

```diff
@@ -1,108 +1,110 @@
 import logging
 import os
+from typing import Any, Generator, List, Union, cast
 
 import pytest
 import structlog
+from structlog.typing import EventDict, WrappedLogger, Processor
 
 try:
     from structlog.contextvars import merge_contextvars
     from structlog.contextvars import clear_contextvars
 except ImportError:
     # structlog < 20.1.0
     # use a "missing" sentinel to avoid a NameError later on
-    merge_contextvars = object()
+    merge_contextvars = lambda *a, **kw: {}  # noqa
     clear_contextvars = lambda *a, **kw: None  # noqa
 
 
 __version__ = "0.6"
 
 
-class EventList(list):
+class EventList(List[EventDict]):
```

Comment:
pyright is not happy with this yet:

```
$ pyright                  
/Users/wim/git/pytest-structlog/tests/test_log.py
  /Users/wim/git/pytest-structlog/tests/test_log.py:90:12 - error: Operator ">=" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
  /Users/wim/git/pytest-structlog/tests/test_log.py:95:12 - error: Operator ">=" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
  /Users/wim/git/pytest-structlog/tests/test_log.py:100:16 - error: Operator ">=" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
  /Users/wim/git/pytest-structlog/tests/test_log.py:105:12 - error: Operator ">=" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
  /Users/wim/git/pytest-structlog/tests/test_log.py:110:16 - error: Operator ">" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
  /Users/wim/git/pytest-structlog/tests/test_log.py:127:12 - error: Operator ">" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
  /Users/wim/git/pytest-structlog/tests/test_log.py:134:12 - error: Operator ">=" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
  /Users/wim/git/pytest-structlog/tests/test_log.py:135:12 - error: Operator ">=" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
  /Users/wim/git/pytest-structlog/tests/test_log.py:136:12 - error: Operator ">=" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
  /Users/wim/git/pytest-structlog/tests/test_log.py:141:16 - error: Operator ">=" not supported for types "EventList" and "list[dict[str, str]]" (reportOperatorIssue)
10 errors, 0 warnings, 0 informations 
```

Can that be improved?

Reply:
1298596536f268357d528f7aed6f623f62ec1f64 adds explicit pyright testing, and replaces the arguments of EventList with `Sequence` not `List`, which allows for covariance i.e in this case because the `dict[str, str]` can be cleanly mapped to `dict[str, Any]`